packages/browser/src/integrations/httpclient.ts

```diff
@@ -0,0 +1,431 @@
+        }
+
+        requestHeaders = headers;
+      }
+
+      const event = this._createEvent({
+        url: xhr.responseURL,
+        method: method,
+        status: xhr.status,
+        requestHeaders,
+        // Can't access request cookies from XHR
+        responseHeaders,
+        responseCookies,
+      });
+
+      // Note: Not adding request / response objects as hints for XHR
+      captureEvent(event);
+    }
+  }
+
+  /**
+   * Extracts response size from `Content-Length` header when possible
+   *
+   * @param headers
+   * @returns The response size in bytes or undefined
+   */
+  private _getResponseSizeFromHeaders(headers?: Record<string, string>): number | undefined {
+    if (headers) {
+      const contentLength = headers['Content-Length'] || headers['content-length'];
+
+      if (contentLength) {
+        return parseInt(contentLength, 10);
+      }
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Creates an object containing cookies from the given cookie string
+   *
+   * @param cookieString The cookie string to parse
+   * @returns The parsed cookies
+   */
+  private _parseCookieString(cookieString: string): Record<string, string> {
+    return cookieString.split('; ').reduce((acc: Record<string, string>, cookie: string) => {
+      const [key, value] = cookie.split('=');
+      acc[key] = value;
+      return acc;
+    }, {});
+  }
+
+  /**
+   * Extracts the headers as an object from the given Fetch API request or response object
+   *
+   * @param headers The headers to extract
+   * @returns The extracted headers as an object
+   */
+  private _extractFetchHeaders(headers: Headers): Record<string, string> {
+    const result: Record<string, string> = {};
+
+    headers.forEach((value, key) => {
+      result[key] = value;
+    });
+
+    return result;
+  }
+
+  /**
+   * Extracts the response headers as an object from the given XHR object
+   *
+   * @param xhr The XHR object to extract the response headers from
+   * @returns The response headers as an object
+   */
+  private _getXHRResponseHeaders(xhr: XMLHttpRequest): Record<string, string> {
+    const headers = xhr.getAllResponseHeaders();
+
+    if (!headers) {
+      return {};
+    }
+
+    return headers.split('\r\n').reduce((acc: Record<string, string>, line: string) => {
+      const [key, value] = line.split(': ');
+      acc[key] = value;
+      return acc;
+    }, {});
+  }
+
+  /**
+   * Checks if the given target url is in the given list of targets
+   *
+   * @param target The target url to check
+   * @returns true if the target url is in the given list of targets, false otherwise
+   */
+  private _isInGivenRequestTargets(target: string): boolean {
+    if (!this._options.failedRequestTargets) {
+      return false;
+    }
+
+    return this._options.failedRequestTargets.some((givenRequestTarget: HttpRequestTarget) => {
+      if (typeof givenRequestTarget === 'string') {
+        return target === givenRequestTarget;
```

Comment:
This should be `return target.includes(givenRequestTarget)`. This option shouldn't require exact matches.